src/senaite/core/content/sampletemplate.py

```diff
@@ -0,0 +1,635 @@
+                default=u"Sample collected by the laboratory"),
+        description=_(u"description_sampletemplate_sampling_required",
+                      default=u"Enable sampling workflow for the created "
+                              u"samples"),
+        default=False,
+        required=False)
+
+    # Partitions (Data Grid)
+    directives.widget(
+        "partitions",
+        DataGridWidgetFactory,
+        allow_insert=False,  # only auto append
+        allow_delete=True,
+        allow_reorder=True,
+        auto_append=True)
+    partitions = schema.List(
+        title=_(u"label_sampletemplate_partitions",
+                default=u"Partitions"),
+        description=_(u"description_sampletemplate_partitions",
+                      default=u"Each line defines a new partition "
+                      u"identified by the 'Partition ID'"),
+        value_type=DataGridRow(
+            title=u"Partition Scheme",
+            schema=IPartitionRecord),
+        required=True,
+        default=[])
+
+    # Auto Partition
+    auto_partition = schema.Bool(
+        title=_(u"title_sampletemplate_auto_partition",
+                default=u"Auto-partition on sample reception"),
+        description=_(u"description_sampletemplate_auto_partition",
+                      default=u"Automatically redirect the user to the "
+                              u"partitions view when the sample is received"),
+        default=False,
+        required=False)
+
+    # Services
+    directives.widget("services",
+                      ListingWidgetFactory,
+                      listing_view="sampletemplate_services_widget")
+    services = schema.List(
+        title=_(
+            u"title_sampletemplate_services",
+            default=u"Services"
+        ),
+        description=_(
+            u"description_sampletemplate_services",
+            default=u"Select the services for this template"
+        ),
+        value_type=DataGridRow(schema=IServiceRecord),
+        default=[],
+        required=True,
+    )
+
+
+@implementer(ISampleTemplate, ISampleTemplateSchema, IDeactivable)
+class SampleTemplate(Container, ClientAwareMixin):
+    """SampleTemplate
+    """
+    # Catalogs where this type will be catalogued
+    _catalogs = [SETUP_CATALOG]
+
+    security = ClassSecurityInfo()
+
+    @security.protected(permissions.View)
+    def getRawSamplePoint(self):
+        accessor = self.accessor("samplepoint", raw=True)
+        return accessor(self)
+
+    @security.protected(permissions.View)
+    def getSamplePoint(self):
+        samplepoint = self.getRawSamplePoint()
+        if not samplepoint:
+            return None
+        return api.get_object(samplepoint)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setSamplePoint(self, value):
+        mutator = self.mutator("samplepoint")
+        mutator(self, value)
+
+    @security.protected(permissions.View)
+    def getSamplePointUID(self):
```

Comment:
Not used anywhere. Remove or flag as `@depreacted("Use getRawSamplePoint instead")`